services/virtualmachine/local_method.go

```diff
@@ -0,0 +1,47 @@
+package virtualmachine
+
+import (
+	"github.com/orbs-network/orbs-spec/types/go/primitives"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+)
+
+func (s *service) getRecentBlockHeight() (primitives.BlockHeight, primitives.TimestampNano, error) {
+	output, err := s.stateStorage.GetStateStorageBlockHeight(&services.GetStateStorageBlockHeightInput{})
+	if err != nil {
+		return 0, 0, err
+	}
+	return output.LastCommittedBlockHeight, output.LastCommittedBlockTimestamp, nil
+}
+
+func (s *service) runLocalMethod(
+	blockHeight primitives.BlockHeight,
+	transaction *protocol.Transaction,
+) (protocol.ExecutionResult, []*protocol.MethodArgument, error) {
+
+	// create execution context
+	contextId, executionContext := s.contexts.allocateExecutionContext(blockHeight, protocol.ACCESS_SCOPE_READ_ONLY)
+	defer s.contexts.destroyExecutionContext(contextId)
+	executionContext.serviceStackPush(transaction.ContractName())
```

Comment:
This line (25) is not covered by any test. From reading the code and tests it seems like this Push is redundant, but it's not. So we need a clear test to see it.